datumaro/components/media.py

```diff
@@ -217,49 +247,171 @@ def size(self) -> Optional[Tuple[int, int]]:
         return self._size
 
     @property
-    def ext(self) -> str:
-        """Media file extension"""
-        if self._ext is not None:
-            return self._ext
+    def ext(self) -> Optional[str]:
+        """Media file extension (with the leading dot)"""
+        return self._ext
+
+    def _get_ext_to_save(self, fp: Union[str, io.IOBase], ext: Optional[str] = None):
+        if isinstance(fp, str):
+            assert ext is None, "'ext' must be empty if string is given."
+            ext = osp.splitext(osp.basename(fp))[1].lower()
         else:
-            return osp.splitext(osp.basename(self.path))[1]
+            ext = ext if ext else self._DEFAULT_EXT
+        return ext
 
     def __eq__(self, other):
+        # Do not compare `_type`
+        # sicne Image is subclass of RoIImage and MosaicImage
         if not isinstance(other, __class__):
             return False
-        return (
-            (np.array_equal(self.size, other.size))
-            and (self.has_data == other.has_data)
-            and (self.has_data and np.array_equal(self.data, other.data) or not self.has_data)
-        )
+        return (np.array_equal(self.size, other.size)) and (np.array_equal(self.data, other.data))
 
-    def save(self, path, crypter: Crypter = NULL_CRYPTER):
-        cur_path = osp.abspath(self.path)
-        path = osp.abspath(path)
+    def set_crypter(self, crypter: Crypter):
+        super().set_crypter(crypter)
 
-        cur_ext = self.ext.lower()
-        new_ext = osp.splitext(osp.basename(path))[1].lower()
 
-        os.makedirs(osp.dirname(path), exist_ok=True)
-        if cur_ext == new_ext and osp.isfile(cur_path):
-            copyto_image(
-                src_path=cur_path, dst_path=path, src_crypter=self._crypter, dst_crypter=crypter
-            )
+class ImageFromFile(FromFileMixin, Image):
+    def __init__(
+        self,
+        path: str,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(path, *args, **kwargs)
+        self.__data = lazy_image(self.path, crypter=self._crypter)
+
+        # extension form file name and real extension can be differ
```

Comment:
```suggestion
        # extension from file name and real extension can be differ
```

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/921/commits/e19ad25b3abdf7c8103198a671e502e1f20d71d3